test/e2e/scheduling/predicates.go

```diff
@@ -831,7 +828,10 @@ var _ = SIGDescribe("SchedulerPredicates [Serial]", func() {
 			framework.Logf("update deployment %q to run rolling upgrade", deploymentName)
 			// change label to kick the rolling upgrade.
 			d.Spec.Template.Labels["test"] = "matchLabelKey2"
-			d, err = cs.AppsV1().Deployments(ns).Update(ctx, newDeployment, metav1.UpdateOptions{})
+			err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+				d, err = cs.AppsV1().Deployments(ns).Update(ctx, newDeployment, metav1.UpdateOptions{})
```

Comment:
For update(), it doesn't work to simply wrap into Retry() (Patch() works btw). Here you have to re-obtain the Deployment upon Conflict error. An example can be found https://github.com/kubernetes/kubernetes/blob/a9593d634c6a053848413e600dadbf974627515f/pkg/controller/namespace/deletion/namespaced_resources_deleter.go#L243

Reply:
👍 
Thanks, will change it and confirm the e2e's running well again in local